docs/how-to/create_plugin.rst

```diff
@@ -0,0 +1,131 @@
+the authentication scheme to use to connect to a remote server - it all
+depends on what your plugin does, and what it needs to do its job! The
+functionality of the plugin is the Python code that makes up its actual
+execution. The functionality of the plugin should take in configuration from
+the plugin's properties.
+
+Create your plugin file
+-----------------------
+The first step in creating a plugin is deciding where it will live. If you're
+adding a plugin directly to ``craft-parts``, then it should go in
+:file:`craft_parts/plugins`. Since we're re-creating the ``dump`` plugin,
+we'll put it in :file:`craft_parts/plugins/new_dump_plugin.py`.
+
+Imports
+~~~~~~~
+First, we'll have to pull in our imports. Since this is such a simple plugin,
+the list of necessary imports is quite small:
+
+.. code:: py
+
+    from typing import Literal
+    from overrides import override
+
+    from .base import Plugin
+    from .properties import PluginProperties
+
+Defining a properties class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+All plugins make use of a "properties" class to describe their unique parameters.
+The dump class is very basic, so we only need to define the required parameters:
+
+.. code:: py
+
+    class NewDumpPluginProperties(PluginProperties, frozen=True):
+        """The part properties used by the new dump plugin."""
+
+        # The human-friendly name of the plugin
+        plugin: Literal["new-dump"] = "new-dump"
+        # The "source" parameter is special - we'll talk about it more soon.
+        source: str
+
+Creating the plugin class
+~~~~~~~~~~~~~~~~~~~~~~~~~
+The plugin class itself contains the functionality of the plugin. Start by
+declaring the properties class:
+
+.. code:: py
+
+    class NewDumpPlugin(Plugin):
+        """Copy the content from the part source."""
+        
+        properties_class = NewDumpPluginProperties
+    
+Next, we'll have to start defining the special methods that make this plugin
+"work". Since this is such a simple plugin, many of these methods are empty:
+
+.. code:: py
+
+    @override
+    def get_build_snaps(self) -> set[str]:
+        """Return a set of required snaps to install in the build
+        environment."""
+        return set()
+
+    @override
+    def get_pull_commands(self) -> list[str]:
+        """Return a list of commands to retrieve dependencies during the
+        pull step."""
+        return []
+
+    @override
+    def get_build_packages(self) -> set[str]:
+        """Return a set of required packages to install in the build
+        environment."""
+        return set()
+
+    @override
+    def get_build_environment(self) -> dict[str, str]:
+        """Return a dictionary with the environment to use in the build
+        step."""
+        return {}
+
+The last method that we have to define, however, is where the actual build
+commands are defined. These are the exact commands executed by a subprocess
```

Comment:
I think it would be a good idea to link the words "build commands" to the page about the parts lifecycle.